gatsby/plugins/gatsby-source-cks/src/node-creation/topics.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Creates Gatsby GraphQL nodes from downloaded data
+ */
+
+import { SourceNodesArgs, NodeInput } from "gatsby";
+import slugify from "slugify";
+
+import { ApiTopicResponse, ApiFullTopic } from "../api/types";
+
+export const topicNodeType = "CksTopic";
+
+export interface TopicNode
+	extends NodeInput,
+		Omit<ApiTopicResponse, "clinicalSpecialties" | "topicHtmlObjects"> {
```

Comment:
This is really nice. Not quite sure on the business rules behind this though but i did see the tests beforehand confirming this functionality.

Reply:
It's not really about business rules, it's more about how we model the data - so using foreign keys to other nodes, rather than sub objects on a single node